### PR TITLE
Allow disabling of scroll-wheel zooming

### PR DESCRIPTION
**Pull request by MichaelJBerk**
When this feature is enabled, scrolling up/down on the trackpad will not zoom in/out of the map

### CODE REVIEW

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3755,6 +3755,8 @@ STR_6304    :Open scenery picker
 STR_6305    :Multithreading
 STR_6306    :{SMALLFONT}{BLACK}Experimental option to use multiple threads to render, may cause instability.
 STR_6307    :Colour scheme: {BLACK}{STRINGID}
+STR_6308    :Trackpad Mode
```
Comment (AaronVanGeffen) [inline data/language/en-GB.txt]: I think the label 'Trackpad Mode' has other, more involved associations for people. Might be best to stick to 'Disable scrolling'.

File: src/openrct2-ui/input/Input.cpp [hunk 2]
```diff
@@ -108,7 +108,9 @@ static void game_handle_key_scroll()
     {
         window_unfollow_sprite(mainWindow);
     }
+    //scrollZOOM
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/input/Input.cpp]: Remove debug comments

**Comment by AaronVanGeffen** [comment]
I'm not particularly sure what problem this intends to solve, but I'm open to the suggestion.

Instead of adding another checkbox, perhaps you could adapt the existing 'zoom to cursor' option into  a dropdown selection. Something like:

```
Scrollwheel behaviour:           [ Zoom to cursor  [▼]]
                                  [Disabled          ]
                                  [Zoom to cursor    ]
                                  [Viewport zoom     ]
```

File: data/language/en-US.txt [hunk 3]
```diff
@@ -924,6 +924,11 @@ STR_6206    :Gray stucco
 STR_6212    :Gray sandstone
 STR_6274    :Can't set color scheme...
 
+STR_6308    :Scroll wheel behavior:
```
Comment (AaronVanGeffen) [inline data/language/en-US.txt]: You should only change the en-UK file on develop. All other language files are updated through our Localisation repository. (If we'd change them here, that would lead to a conflict.)

File: src/openrct2/localisation/StringIds.h [hunk 4]
```diff
@@ -3938,9 +3938,14 @@ enum
...
     STR_TILE_INSPECTOR_COLOUR_SCHEME = 6307,
+    
+    STR_SCROLL_MODE = 6308,
+    STR_SCROLL_MODE_STANDARD = 6309,
+    STR_SCROLL_MODE_DISABLED = 6310,
+    STR_SCROLL_MODE_TIP = 6311,
 
     // Have to include resource strings (from scenarios and objects) for the time being now that language is partially working
-    STR_COUNT = 32768
+    STR_COUNT = 32771
```
Comment (AaronVanGeffen) [inline src/openrct2/localisation/StringIds.h]: Please revert this change.

```suggestion
    STR_COUNT = 32769
```

**Comment by AaronVanGeffen** [comment]
For the record, this is what it looks like at the moment.

Screenshot_20190425_233312

The options should be capitalised consistently. Personally, I prefer them to be not use title case. I also think 'Disabled' should be the first option, rather than the third.

I'd also like

File: src/openrct2-ui/windows/Options.cpp [hunk 5]
```diff
@@ -1869,14 +1906,28 @@ static void window_options_invalidate(rct_window* w)
...
-            widget_set_checkbox_value(w, WIDX_ZOOM_TO_CURSOR, gConfigGeneral.zoom_to_cursor);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_FINANCES, gConfigInterface.toolbar_show_finances);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_RESEARCH, gConfigInterface.toolbar_show_research);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_CHEATS, gConfigInterface.toolbar_show_cheats);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_NEWS, gConfigInterface.toolbar_show_news);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_MUTE, gConfigInterface.toolbar_show_mute);
             widget_set_checkbox_value(w, WIDX_TOOLBAR_SHOW_CHAT, gConfigInterface.toolbar_show_chat);
-
+            
+            if (gConfigGeneral.zoom_to_cursor == false && gConfigGeneral.disable_scroll_zoom == false)
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/windows/Options.cpp]: I'd prefer `zoom_to_cursor` to be replaced with a `scroll_wheel_behaviour` (integer) option, rather than adding another `disable_scroll_zoom` bool. This block can then use a switch or array-based string lookup instead.

File: src/openrct2-ui/interface/Window.cpp [hunk 6]
```diff
@@ -435,9 +435,9 @@ static void window_viewport_wheel_input(rct_window* w, int32_t wheel)
 {
     if (gScreenFlags & (SCREEN_FLAGS_TRACK_MANAGER | SCREEN_FLAGS_TITLE_DEMO))
         return;
-    if (wheel < 0 && !gConfigGeneral.disable_scroll_zoom)
+    if (wheel < 0 && !gConfigGeneral.scroll_wheel_behaviour == 0)
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/interface/Window.cpp]: Did you test this? `!gConfigGeneral.scroll_wheel_behaviour == 0` will bind the `!` operator to `gConfigGeneral.scroll_wheel_behaviour`, inverting the setting rather than the expression.

I'd preferred for the entire block to be nested in one `gConfigGeneral.scroll_wheel_behaviour != 0` if-block.

File: src/openrct2/config/Config.cpp [hunk 7]
```diff
@@ -206,9 +205,9 @@ namespace Config
             model->last_save_landscape_directory = reader->GetCString("last_landscape_directory", nullptr);
             model->last_save_scenario_directory = reader->GetCString("last_scenario_directory", nullptr);
             model->last_save_track_directory = reader->GetCString("last_track_directory", nullptr);
+            model->scroll_wheel_behaviour = reader->GetInt32("scroll_wheel_behaviour", 1);
```
Comment (AaronVanGeffen) [inline src/openrct2/config/Config.cpp]: Zooming to cursor should still be the default. You've assigned this `2`, not `1`.

**Comment by Gymnasiast** [comment]
Like Aaron, I would like to know what problem this pull request aims to solve. We generally only allow disabling functionality if it can cause problems to a significant amount of users. For mousewheel zooming, we haven't had any requests to disable it so far.

Additionally, Aaron highlighted some issues with the code itself, which you neither addressed nor replied to.

**Comment by AaronVanGeffen** [comment]
It's been requested before; @MichaelJBerk mentioned #7905 in his original PR title, which was requested last year. The discussion label was never, though.

I don't object to the change, but I also wonder what problem it's solving. If the touchpad is too sensitive to zooming while panning around, I think the better solution would be to tweak that sensitivity instead.

**Comment by Gymnasiast** [comment]
I don't think this feature is really useful. If a user's trackpad is too sensitive, I don't think OpenRCT2 is the correct place to mitigate it.

Besides that, the code quality of this PR is not quite up to snuff and you didn't react to Aaron's feedback.

Now, closing PRs is not something I do lightly, but I don't want to drag people around and giving them false hope.
